multiplexer/abci/multiplexer.go

```diff
@@ -226,10 +226,7 @@ func (m *Multiplexer) startApp() error {
 	}
 
 	if currentVersion.Appd.Pid() == appd.AppdStopped {
-		programArgs := os.Args
-		if len(os.Args) > 2 {
-			programArgs = os.Args[2:] // remove 'appd start' args
```

Comment:
this was the problem. This assumed that the first two os.Args were:

```
celestia-appd start
```

but that's not always the case as Everstake pointed out.